### PR TITLE
Fixes #730. Focus and blur the search input when opening and closing.

### LINKED ISSUE #730: Forced autofocus when open Search to Homepage
autofocus on input seems not working. @miketaylr  maybe we can forced autofcus with javascript ?

### PR DESCRIPTION
**Pull request by miketaylr**
r? @magsout 

I also noticed that the close button was submitting the form, apparently the default type for `<button>` is `type=submit`. TIL.

### CODE REVIEW

**Comment by miketaylr** [comment]
Blah, ignore the test failures. I'm working on that in #728

File: webcompat/templates/topbar.html [hunk 1]
```diff
@@ -48,7 +48,7 @@
           <span class="wc-Navbar-link-icon wc-Icon wc-Icon--search" aria-hidden="true"></span>
         </div>
         <input class="wc-SearchBar-input" autofocus name="q" placeholder="Search by keyword"/>
-        <button class="r-ResetButton wc-SearchBar-icon wc-SearchBar-icon--button js-SearchBarClose" ara-hidden="true">
+        <button class="r-ResetButton wc-SearchBar-icon wc-SearchBar-icon--button js-SearchBarClose" ara-hidden="true" type="button">
```
Comment (magsout) [inline webcompat/templates/topbar.html]: ah yes good catch, my bad.

**Comment by magsout** [comment]
r+

**Comment by miketaylr** [comment]
Thanks!

**Pull request merged**